### ISSUE TITLE
[Usage]:  How to infer the reward model of nvidia/Llama-3.1-Nemotron-70B-Reward-HF?

### ISSUE THREAD

**Issue by xs1997zju** [vllm-project/vllm#11459]
### Your current environment

```text
The output of `python collect_env.py`
```
python3.9

### How would you like to use vllm

I want to run inference of a [specific model](put link here). I don't know how to integrate it with vllm.

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by xs1997zju** [comment]
here is my infer scripts:
`from vllm import LLM, SamplingParams
from transformers import AutoModelForCausalLM, AutoTokenizer

model_path = 'nvidia/Llama-3.1-Nemotron-70B-Reward-HF'

model = LLM(model=model_path, 
            tensor_parallel_size=8, 
            dtype="float16", 
            enable_chunked_prefill=False, 
            task="embed",
            )

tokenizer = AutoTokenizer.from_pretrained(model_path)

prompt = "What is 1+1?"

good_response = "1+1=2"

bad_response = "1+1=3"

messages = [
    {'role': "user", "content": prompt}, 
    {'role': "assistant", "content": good_response}
]

prompt = tokenizer.apply_chat_template(messages, tokenize=False, add_generation_prompt=False)

output = model.encode(prompt)

# output[0].output.data.size(): [8192]`
but i want get the reward score, which should be: reward for good_response = -3.28125, according to the HF example
<img width="580" alt="image" src="https://github.com/user-attachments/assets/65a1d9b7-eb94-44e0-a0e1-b89dd129cab4" />

**Comment by DarkLight1337** [comment]
See https://github.com/vllm-project/vllm/issues/8967#issuecomment-2560818480

**Comment by DarkLight1337** [comment]
Please try out #11469. You should be able to run the model just by setting `--task reward`.

**Comment by nzw0301** [comment]
Hi, in my understanding, `nvidia/Llama-3.1-Nemotron-70B-Reward-HF` is implemented as a causal LM, and its reward computation uses `lm_lead`  Thus `as_reward_model` does not work for it, since `lm_head` is aways removed https://github.com/vllm-project/vllm/blob/9ddac56311b28f08e40a941296eb66fbb1be0a7a/vllm/model_executor/models/adapters.py#L56-L59. I suppose one workaround is to convert this causal model into a reward model explicitly.

**Comment by DarkLight1337** [comment]
That alone isn't quite enough to solve the

**Comment by ControllableGeneration** [comment]
Try this code snippet to get logits:
`logits = model.lm_head.linear_method.apply(model.lm_head, hidden_states)`

I have verified that the logits here is semi-identical to those obtained by huggingface transformers
